Google.PowerShell/Compute/GceInstanceDescriptionCmdlet.cs

```diff
@@ -190,6 +203,64 @@ protected virtual IList<AttachedDisk> BuildAttachedDisks()
             return disks;
         }
 
+        /// <summary>
+        /// Creates a machine type string. Most of the logic is for the
+        /// custom machine case where -CustomMemory or -CustomCpu is used.
+        /// If nothing is supplied, use "n1-standard-1" as default.
+        /// </summary>
+        /// <returns></returns>
+        protected string BuildMachineType()
+        {
+            if (string.IsNullOrWhiteSpace(MachineType) && !CustomCpu.HasValue
+                && !CustomMemory.HasValue)
+            {
+                return "n1-standard-1";
+            }
+
+            if (!string.IsNullOrWhiteSpace(MachineType))
+            {
+                if (CustomCpu.HasValue || CustomMemory.HasValue)
+                {
+                    throw new PSArgumentException(
+                        "-CustomMemory or -CustomCpu cannot be used with -MachineType.");
+                }
+                return MachineType;
+            }
+
+            if ((CustomCpu.HasValue && !CustomMemory.HasValue)
+                || (!CustomCpu.HasValue && CustomMemory.HasValue))
+            {
+                throw new PSArgumentException(
+                    "-CustomMemory and -CustomCpu must be used together.");
+            }
+
+            if (CustomCpu < 1 || (CustomCpu > 1 && CustomCpu % 2 != 0))
+            {
+                throw new PSArgumentException("Number of custom vCPUs must be positive and be either 1 or even.");
+            }
+
+            if (CustomMemory < 0 || CustomMemory % 256 != 0)
+            {
+                throw new PSArgumentException("Total custom memory must be positive and a multiple of 256.");
+            }
+
+            double memoryPerCpu = (double)CustomMemory/(double)CustomCpu;
+
+            if (memoryPerCpu < 0.9 * AmountOfMbInGb)
```

Comment:
Can this value change? Would it be better to make the request and return the service error message?